tianshou/data/collector.py

```diff
@@ -344,45 +538,136 @@ class Collector(BaseCollector):
     # A - dimension(s) of actions
     # H - dimension(s) of hidden state
     # D - number of envs that reached done in the current collect iteration. Only relevant in n_episode case.
-    # S - number of surplus envs, i.e. envs that are ready but won't be used in the next iteration.
+    # S - number of surplus envs, i.e., envs that are ready but won't be used in the next iteration.
     #     Only used in n_episode case. Then, R becomes R-S.
+    # local_index - selecting from the locally available environments. In more details:
+    #     Each env is associated to a number in [0,..., N-1]. At any moment there are R ready envs,
+    #     but they are not necessarily equal to [0, ..., R-1]. Let the R corresponding indices be
+    #     [r_0, ..., r_(R-1)] (each r_i is in [0, ... N-1]). If the local index is
+    #     [0, 1, 2], it means that we want to select envs [r_0, r_1, r_2].
+    #     We will usually select from the ready envs by slicing like `ready_env_idx_R[local_index]`
+    # global_index - the index in [0, ..., N-1]. Slicing the a `_R` index by a local_index produces the
+    #     corresponding global index. In the example above:
+    #     1. _R index is [r_0, ..., r_(R-1)]
+    #     2. local_index is [0, 1, 2]
+    #     3. global_index is [r_0, r_1, r_2] and can be used to select from an array of length N
+    #
     def __init__(
         self,
         policy: BasePolicy,
         env: gym.Env | BaseVectorEnv,
         buffer: ReplayBuffer | None = None,
         exploration_noise: bool = False,
+        on_episode_done_hook: Optional["EpisodeRolloutHookProtocol"] = None,
+        on_step_hook: Optional["StepHookProtocol"] = None,
+        raise_on_nan_in_buffer: bool = True,
```

Comment:
Argument is unused

Reply:
Nice catch! I moved the check to BaseCollector now